packages/components/src/DateTimePickers/Time/Manager/Manager.component.js

```diff
@@ -1,84 +1,78 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import PropTypes from 'prop-types';
 
 import { TimeContext } from '../Context';
 import extractTime, { getTimeFormat } from '../time-extraction';
 
-class ContextualManager extends React.Component {
-	static displayName = 'Time.Manager';
-	static propTypes = {
-		children: PropTypes.node,
-		onChange: PropTypes.func,
-		useSeconds: PropTypes.bool,
-		timezone: PropTypes.string,
-		value: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
-	};
+function TimeContextualManager(props) {
+	const [state, setState] = useState(extractTime(props.value, props.useSeconds));
 
-	static defaultProps = {
-		useSeconds: false,
-	};
-
-	constructor(props) {
-		super(props);
-
-		this.state = extractTime(props.value, props.useSeconds);
-		this.onChange = this.onChange.bind(this);
-		this.onInputChange = this.onInputChange.bind(this);
-		this.onPickerChange = this.onPickerChange.bind(this);
-	}
+	useEffect(() => {
+		const nextState = extractTime(props.value, props.useSeconds);
+		if (!nextState.errorMessage) {
+			setState(nextState);
+		}
+	}, [props.value, props.useSeconds]);
 
-	onChange(event, origin) {
-		if (!this.props.onChange) {
+	function onChange(event, origin, nextState) {
+		if (!props.onChange) {
 			return;
 		}
-		this.props.onChange(event, { ...this.state, origin });
+		props.onChange(event, { ...nextState, origin });
 	}
 
-	onInputChange(event) {
+	function onInputChange(event) {
 		const textInput = event.target.value;
-		const nextState = extractTime(textInput, this.props.useSeconds);
-		this.setState(nextState, () => {
-			this.onChange(event, 'INPUT');
-		});
+		const nextState = extractTime(textInput, props.useSeconds);
+		setState(nextState);
```

Comment:
You can put that `setState` in `onChange()` instead, and remove this line and the one in `onPickerChange()`.
That's not a big deal, but just a suggestion :)

(I like the idea that we have a common `onChange`, and 2 specific change fonctions, that only differs in the way to create the data we save)

Reply:
thanks :) suggestions applied.